### PR TITLE
pre-release: ASI upgrade

### PR DESCRIPTION
**Pull request by MissingNO57**
## Proposed changes

- Changed default prefix to `asi`
- Added Eridanus configuration
- Changed all `fetch` addresses to `asi` addresses in examples, tests, and docs.

Do we know new chain-id for `fetchhub-4` ? 

## Issues

Links to any issues resolved.

## Types of changes

What types of changes does your code introduce?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to stop working as expected)
- [X] Something else (e.g. tests, scripts, example, deployment, infrastructure, ...)

## Checklist

_Put an `x` in the boxes that apply._

- [X] I have read the [CONTRIBUTING](https://github.com/fetchai/cosmpy/blob/main/CONTRIBUTING.md) document.
- [X] I have based my branch, and I am making a pull request against, the `main` branch.
- [X] Checks and tests pass locally.

### If applicable

- [X] I have added tests that prove my fix is effective or that my feature works.
- [X] I have checked that code coverage does not decrease.
- [X] I have added/updated the documentations.

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did, what alternatives you considered, etc.

### CODE REVIEW

File: tests/integration/test_contract.py [hunk 1]
```diff
@@ -177,12 +177,12 @@ class TestContractRestAPI(TestContract):
 
     def _get_network_config(self):
```
Comment (MissingNO57) [inline tests/integration/test_contract.py:178]: In some integration tests we deploy sample contracts on testnet in every run! This leads to adding unnecessary data there.

**Review by jrriehl — APPROVED** [review]
LGTM!

File: cosmpy/aerial/config.py [hunk 2]
```diff
@@ -125,8 +140,8 @@ def fetchai_mainnet(cls) -> "NetworkConfig":
             chain_id="fetchhub-4",
             url="grpc+https://grpc-fetchhub.fetch.ai",
             fee_minimum_gas_price=0,
```
Comment (pbukva) [inline cosmpy/aerial/config.py:142]: Note: We have been considering setting this default minimum gas price here to non-zero value (the same value as for testnet). Even though this is not enforceable, practically speaking, this is what we want - if nothing else, then at least  defaulting to non-zero Tx gas price value will make it less straightforward for users to use zero Tx fee.

File: cosmpy/aerial/config.py [hunk 3]
```diff
@@ -125,8 +140,8 @@ def fetchai_mainnet(cls) -> "NetworkConfig":
             chain_id="fetchhub-4",
```
Comment (pbukva) [inline cosmpy/aerial/config.py:140]:
```suggestion
            chain_id="asi-1",
```

File: cosmpy/aerial/config.py [hunk 4]
```diff
@@ -125,8 +140,8 @@ def fetchai_mainnet(cls) -> "NetworkConfig":
...
             url="grpc+https://grpc-fetchhub.fetch.ai",
```
Comment (pbukva) [inline cosmpy/aerial/config.py:141]:
```suggestion
            url="grpc+https://grpc-asi.fetch.ai",
```

File: cosmpy/aerial/config.py (hunk 2 above)
Comment (pbukva) [reply cosmpy/aerial/config.py:142]: For now (for ASI umgrade release), please ignore my comment above. We will touch on thsese kind of things in later releases.

Comment (pbukva) [reply cosmpy/aerial/config.py:142]: Note: In this sense (= not now), we shall also extend the `NetworkConfig` class with address prefix data member.

**Review by pbukva — APPROVED** [review]
LGTM

**Review by pbukva — APPROVED** [review]
LGTM

**Pull request merged**